### PR TITLE
enable streaming input of multiple audio streams in cacheawarestreamingaudiobuffer

### PR DESCRIPTION
**Pull request by shababo**
# What does this PR do ?

Enables `CacheAwareStreamingAudioBuffer` to handle streaming input by calling `append_audio` directly and calling `get_next_chunk`.

**Collection**: [Note which collection this PR will affect]

ASR

# Changelog

- Add specific line by line info of high level changes in this PR.

Call `append_audio` with a chunk of audio bytes to append the preprocessed feature buffer.
Call `[get_next_chunk](https://github.com/modal-projects/NeMo/blob/shababo/add-streaming-input-to-cacheawarestreamingaudiobuffer/nemo/collections/asr/parts/utils/streaming_utils.py#L1595-L1596)` to get the next full sized feature chunk.

# Usage

- You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
async def transcribe(self, audio_data, step_num, stream_id = -1) -> str:

    print(f"transcribing audio data: {len(audio_data)} bytes")

    drop_extra_pre_encoded = (
        0
        if step_num == 0 and not self.cfg.pad_and_drop_preencoded
        else self.asr_model.encoder.streaming_cfg.drop_extra_pre_encoded
    )
    # convert to numpy
    audio_data = int2float(np.frombuffer(audio_data, dtype=np.int16))
    processed_signal, processed_signal_length, stream_id = self.streaming_buffer.append_audio(audio_data, stream_id=stream_id)
    
    result = self.streaming_buffer.get_next_chunk()
    if result is not None:
        audio_chunk, chunk_lengths = result
    else:
        return None, stream_id

    with torch.inference_mode():
        with torch.amp.autocast(self.diar_model.device.type, enabled=True):
            with torch.no_grad():
                result = self.multispk_asr_streamer.perform_parallel_streaming_stt_spk(
                    step_num=step_num,
                    chunk_audio=audio_chunk,
                    chunk_lengths=chunk_lengths,
                    is_buffer_empty=False,
                    drop_extra_pre_encoded=drop_extra_pre_encoded,
                )
    if result:
        return result[0], stream_id
    return None, stream_id
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

**Comment by shababo** [comment]
@weiqingw4ng here's the new PR!
